x-pack/solutions/observability/plugins/slo/public/pages/slo_details/components/historical_data_charts.tsx

```diff
@@ -45,14 +38,30 @@ export function HistoricalDataCharts({
   );
   const historicalSliData = formatHistoricalData(sloHistoricalSummary?.data, 'sli_value');
 
+  // Calculate observed value from the latest entry in historical data within the time range
+  // Historical data is already filtered by the time range when fetched
+  const observedValue = React.useMemo(() => {
```

Comment:
We can remove useMemo. Also, just take the latest value of the summary data, it's already sorted by the backend.